src/System.Windows.Forms/src/System/Windows/Forms/Control.cs

```diff
@@ -11787,10 +11787,13 @@ private void UpdateBindings()
         ///  reflect its index.
         /// </summary>
         private void UpdateChildControlIndex(Control control)
-        {
-            // Don't reorder the child control array for tab controls. Implemented as a special case
-            // in order to keep the method private.
-            if (GetType().IsAssignableFrom(typeof(TabControl)))
+        {            
+            if (// Don't reorder the child control array for tab controls. Implemented as a special case
+                // in order to keep the method private.
+                this is TabControl ||
+                // Also short-circuit when the Control class is instantiated directly. This is to provide
+                // consistency with the behavior prior to bug fix https://github.com/dotnet/winforms/issues/7837
+                GetType() == typeof(Control))
```

Comment:
Please move comments above the if
```suggestion
            // Don't reorder the child control array for tab controls. Implemented as a special case
            // in order to keep the method private.
            // Also short-circuit when the Control class is instantiated directly. This is to provide
            // consistency with the behavior prior to bug fix https://github.com/dotnet/winforms/issues/7837
            if (this is TabControl || GetType() == typeof(Control))
```

Reply:
Done

Reply:
This is still outstanding. Let's move both comments above the if and format if statement like suggestion. Everything else looks good so once that is fixed we can get this merged :)